samza-core/src/main/scala/org/apache/samza/system/SystemConsumers.scala

```diff
@@ -112,13 +114,22 @@ class SystemConsumers (
    * Clock can be used to inject a custom clock when mocking this class in
    * tests. The default implementation returns the current system clock time.
    */
-  val clock: () => Long = () => System.nanoTime()) extends Logging with TimerUtil {
+  val clock: () => Long = () => System.nanoTime(),
+
+  val elasticityFactor: Int = 1) extends Logging with TimerUtil {
 
   /**
    * Mapping from the {@see SystemStreamPartition} to the registered offsets.
    */
   private val sspToRegisteredOffsets = new HashMap[SystemStreamPartition, String]()
 
+  /**
+   * Set of all the SystemStreamPartitions registered with this SystemConsumers
+   * With elasticity enabled, the SystemStreamPartitions are actually key buckets within a full SSP
+   * Without elasticity, there are no key buckets and hence is the full SSP
```

Comment:
Nit: 
With elasticity-enabled, the SSPs have valid (i.e. >=0) keyBuckets, with 
elasticity disabled keyBuckets on all SSPs are = -1.